onadata/apps/api/tests/viewsets/test_open_data_viewset.py

```diff
@@ -110,6 +110,43 @@ def test_get_data_using_uuid(self):
         # cast generator response to list so that we can get the response count
         self.assertEqual(len(streaming_data(response)), 4)
 
+    def test_column_headers_match_row_heaaders(self):
+        """
+        Test that the row headers generated
+        match the column headers for the same form
+        """
+        self.view = OpenDataViewSet.as_view({
+            'get': 'schema'
+        })
+
+        _open_data = self.get_open_data_object()
+        uuid = _open_data.uuid
+        request1 = self.factory.get('/', **self.extra)
+        response1 = self.view(request1, uuid=uuid)
+        self.assertEqual(response1.status_code, 200)
+        self.assertListEqual(
+            ['column_headers', 'connection_name', 'table_alias'],
+            sorted(list(response1.data))
+        )
+        expected_column_header_fields = [
+            k['id'] for k in response1.data['column_headers']]
+
+        self._make_submissions()
+        self.view = OpenDataViewSet.as_view({
+            'get': 'data'
+        })
+        _open_data = self.get_open_data_object()
+        uuid = _open_data.uuid
+        request2 = self.factory.get('/', **self.extra)
+        response2 = self.view(request2, uuid=uuid)
+        self.assertEqual(response2.status_code, 200)
+        # cast generator response to list for easy manipulation
+        row_data = streaming_data(response2)
+        row_data_fields = [k for k in row_data[2].keys()]
+        self.assertEqual(
+            row_data_fields.sort(),
+            expected_column_header_fields.sort())
```

Comment:
This test assumes the first response(`response1`) is always what we should be expecting to be correct ? 

_Could we change the `expected_headers` to be a static dict `['header1', 'header2']` to make this less dynamic ? I feel like this test will never fail...._

Reply:
Sure. changed that.